sdks/java/core/src/main/java/org/apache/beam/sdk/util/construction/Environments.java

```diff
@@ -219,20 +219,16 @@ public static Environment createDockerEnvironment(String dockerImageUrl) {
   }
 
   private static Environment createExternalEnvironment(String externalServiceAddress) {
-    if (externalServiceAddress.isEmpty()) {
-      throw new IllegalArgumentException(
-          String.format(
-              "External service address must not be empty (set it using '--environmentOptions=%s=...'?).",
-              externalServiceAddressOption));
+    // Create the payload builder. If the address is empty, the payload will be empty,
+    // acting as a placeholder for late binding (e.g., for LOOPBACK environment).
```

Comment:
> as a placeholder for late binding

nit: for reference, would be nice to leave a code pointer where later binding happens (if already figured out).

Reply:
Updated the comments to mention LOOPBACK.